integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/util/ApictlUtils.java

```diff
@@ -262,22 +276,27 @@ public static void deployAPI(String apiProjectName, String mgwEnv) throws MicroG
      * @param apiName name of the API (in api.yaml) to undeploy
      * @param apiVersion version of the API
      * @param mgwEnv name of the apictl mgw env the API was deployed
+     * @param vhost vhost of the API to be undeployed from
      * @throws MicroGWTestException if apictl was unable to undeploy the API
      */
-    public static void undeployAPI(String apiName, String apiVersion, String mgwEnv) throws MicroGWTestException {
+    public static void undeployAPI(String apiName, String apiVersion, String mgwEnv, String vhost) throws MicroGWTestException {
         String[] cmdArray = { MG, UNDEPLOY, API };
-        String[] argsArray = { NAME_FLAG, apiName, VERSION_FLAG, apiVersion, ENV_FLAG, mgwEnv };
+        List<String> args = new ArrayList<>(Arrays.asList(NAME_FLAG, apiName, VERSION_FLAG, apiVersion, ENV_FLAG, mgwEnv));
+        if (StringUtils.isNotEmpty(vhost)) {
+            args.addAll(Arrays.asList(VHOST_FLAG, vhost));
+        }
+        String[] argsArray = args.toArray(String[]::new);
         try {
             String[] responseLines = runApictlCommand(cmdArray, argsArray, 1);
             if (responseLines[0]!= null && !responseLines[0].startsWith(SUCCESSFULLY_UNDEPLOYED_RESPONSE)) {
                 throw new MicroGWTestException("Unable to undeploy API: "
-                        + apiName + " from microgateway adapter environment: " + mgwEnv);
+                        + apiName + " from microgateway adapter environment: " + mgwEnv + " vhost: " + vhost);
             }
         } catch (IOException e) {
             throw new MicroGWTestException("Unable to undeploy API: "
                     + apiName + " to microgateway adapter environment: " + mgwEnv, e);
```

Comment:
```suggestion
                    + apiName + " from microgateway adapter environment: " + mgwEnv + " vhost: "+ vhost, e);
```
might have to check what happens when vhost is null

Reply:
fixed with https://github.com/wso2/product-microgateway/pull/1949/commits/de3529307cac69e02b7088edd6c868fe7ab356b7